docs/dev/release_process.rst

```diff
@@ -39,6 +39,31 @@ Create a release branch
 If this is a new major or minor release, you need to make a new branch as described above.
 
 
+Update any translation files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If string interface text has changed, or more complete translations are available, translation files should be updated.
+This is currently done by running the ``make downloadmessages`` command. Following this, the specific files that have been updated with approved translations will need to be added to the repository.
+
+Caveat: by default Crowdin will download all translations, not just approved ones, and will often download untranslated strings also. Do not just add all the files that are downloaded when ``make downloadmessages`` is run, as this will lead to untranslated and poor quality strings being included.
+
+If you need to add a new interface language to Kolibri, please see :ref:`new_language` for details.
+
+Finally, strings for any external Kolibri plugins (like kolibri-exercise-perseus-renderer) should also have been updated, a new release made, and the version updated in Kolibri.
+
+
+Squash migrations
+~~~~~~~~~~~~~~~~~
+
+When possible, we like to utilize the Django migration squashing to simplify the migration path for new users (while simultaneously maintaining the migration path for old users). So far this has not been done, due to the existence of data migrations in our migration history. Once we have upgraded to Django 1.11, we will be able to mark these data migrations as elidable, and we will be able to better squash our history.
+
+
+Ensure bugfixes from internal depencies have propagated
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some issues in Kolibri arise due to our integration of internally produced, but external to Kolibri, packages, such as kolibri-exercise-perseus-renderer, iceqube, and morango. If any of these kinds of dependencies have been updated to fix issues for this milestone, then the dependency version should have been updated.
+
```

Comment:
Maybe another section like this:

> ### Update the changelog
> 
> In general we should try to keep the changelog up-to-date as PRs are merged in; however in practice the changelog usually needs to be cleaned up, fleshed out, and clarified.
> 
> Our changelogs should represent list:
> * significant new features that were added
> * significant categories of bug fixes or user-facing improvements
> * significant behind-the-scenes technical improvements
>  
> Keep entries concise and consistent with the established writing style. The changelog should _not_ include an entry for every PR or every issue closed. Reading the changelog should give a quick, high-level, semi-technical summary of what has changed.
> 
> Note that for older patch releases, the change should only be mentioned once: it is implied that fixes in older releases are propagated forward.

Also, in the `Tag the release` section, we can say:

> Copy the entries from the changelog into github's "Release notes"

Reply:
Added.